src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,520 @@
+import org.openstreetmap.atlas.utilities.conversion.StringConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.io.WKBReader;
+
+/**
+ * Configurable {@link AtlasEntity} filter that passes through anything that intersects with
+ * includePolygons polygons or anything that doesn't intersect with exclude polygons. Exclude
+ * polygons are looked at only if no includePolygons polygons exist. Include takes precedence and
+ * polygons intersecting with previously read polygons are dropped with a warning.
+ * 
+ * @author jklamer
+ */
+public final class AtlasEntityPolygonsFilter implements Predicate<AtlasEntity>, Serializable
+{
+    public static final String EXCLUDED_MULTIPOLYGONS_KEY = "filter.multipolygons.exclude";
+    public static final String EXCLUDED_POLYGONS_KEY = "filter.polygons.exclude";
+    public static final String INCLUDED_MULTIPOLYGONS_KEY = "filter.multipolygons.include";
+    public static final String INCLUDED_POLYGONS_KEY = "filter.polygons.include";
+    private static final Logger logger = LoggerFactory.getLogger(AtlasEntityPolygonsFilter.class);
+    public static final IntersectionDeciding DEFAULT_INTERSECTION_DECIDING = new IntersectionDeciding()
+    {
+        @Override
+        public boolean multiPolygonEntityIntersecting(final MultiPolygon multiPolygon,
+                final AtlasEntity entity)
+        {
+            if (entity instanceof LineItem)
+            {
+                return multiPolygon.overlaps(((LineItem) entity).asPolyLine());
+            }
+            if (entity instanceof LocationItem)
+            {
+                return multiPolygon
+                        .fullyGeometricallyEncloses(((LocationItem) entity).getLocation());
+            }
+            if (entity instanceof Area)
+            {
+                return multiPolygon.overlaps(((Area) entity).asPolygon());
+            }
+            if (entity instanceof Relation)
+            {
+                return ((Relation) entity).members().stream().map(RelationMember::getEntity)
+                        .anyMatch(relationEntity -> this
+                                .multiPolygonEntityIntersecting(multiPolygon, relationEntity));
+            }
+            else
+            {
+                logger.warn("Unknown AtlasEntity Implementation {}", entity);
+                return false;
+            }
+        }
+
+        @Override
+        public boolean polygonEntityIntersecting(final Polygon polygon, final AtlasEntity entity)
+        {
+            return entity.intersects(polygon);
+        }
+    };
+    private static final long serialVersionUID = -3474748398986569205L;
+    private List<MultiPolygon> excludeMultiPolygons = new ArrayList<>();
+    private List<Polygon> excludePolygons = new ArrayList<>();
+    private Type filterType;
+    private List<MultiPolygon> includeMultiPolygons = new ArrayList<>();
+    private List<Polygon> includePolygons = new ArrayList<>();
```

Comment:
Why do we separate the polygons and multipolygons into separate collections? What if use `Iterable<Polygon>` to combine both into one.

Reply:
The ```Iterable<Polygon>``` of polygons returned by a ```MultiPolygon``` are not equal is the sense of intersection logic because they are both outers and inners. Also, the ```MultiPolygon``` objects has a lot of useful methods when it comes to determining intersection considering both its inner and outer rings